### PR TITLE
Create admin pages in rails

### LINKED ISSUE #77: Create Admin page(s) in rails
Create Login page where admin can log in and after login, admin can land up on the home page where admin can create articles and URLs.

@sudeeptarlekar _a

### PR DESCRIPTION
**Pull request by sudeeptarlekar**
This fixes #77 

@vipulnsward _a please review

### CODE REVIEW

**Comment by sudeeptarlekar** [comment]
@sudeeptarlekar _a push secret.yml file

File: app/models/concerns/user_role_concern.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module UserRoleConcern
```
[the remark this answers is not in the document]
Comment (sudeeptarlekar) [reply app/models/concerns/user_role_concern.rb]: Right now we are not using this, my thought process was down the line we will need it when multiple organizations will create their accounts, but anyway we can add this later. Removing for now.

[the remark this answers is not in the document]
Comment (vipulnsward) [reply app/models/concerns/user_role_concern.rb]: Please never overthink and more things then needed. Its one of the reasons we don't keep commented code in files. We can always add it back later when needed.

**Comment by vipulnsward** [comment]
@rkr090 _a Can you review it once?

**Comment by vipulnsward** [comment]
Sorry, meaning whenever its ready for review.

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+  # ==> Controller configuration
+  # Configure the parent class to the devise controllers.
+  # config.parent_controller = 'DeviseController'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (rkr090) [inline config/initializers/devise.rb]: Let's move this to 'secrets.yml' ```Rails.application.secrets.mailer_default_from_email```

The from email can be ```"'Learn Academy' <do-not-reply@acehelp.com>"```

Note: ```Learn Academy``` is the title on the application layout.

File: config/secrets.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  devise:
+    secret_key: 0d1067153029e34a519dcf73db39b77504b7c3c856673ff0f361d9c5eeca1cfd28d639692d55d35dda28bc265195a87f3b3e42f4947c8f4c144723a8f0866120
+  api_base_url: 'http://staging.acehelp.com/api/v1'
+
+production:
+  store_uploaded_assets_in: aws_s3
+  secret_key_base: <%= ENV['SECRET_KEY_BASE'] %>
+  devise:
+    secret_key: <%= ENV['DEVISE_KEY'] %>
+  api_base_url: ''
```
Comment (rkr090) [inline config/secrets.yml]: IMO, The tests are failing because there are no secrets in the test environment.

File: test/models/user_test.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (rkr090) [inline test/models/user_test.rb]: @sudeeptarlekar Lets also add user fixtures and tests

File: app/controllers/home_controller.rb [hunk 5]
```diff
@@ -2,7 +2,11 @@
 
 class HomeController < ApplicationController
   def index
-    render
+    if current_user
+      redirect_to admin_dashboard_index_path
```
Comment (rkr090) [inline app/controllers/home_controller.rb]: Every user can't be redirected to the admin dashboard. We need to identify admins using a flag ```super_admin``` on the user's table.

We can add ```roles``` later as the need arises. cc: @vipulnsward

File: config/routes.rb [hunk 6]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  devise_for :users
```
Comment (rkr090) [inline config/routes.rb]: One we have a way of identifying admins vs regular users, we will need to add **authenticated** and **unauthenticated** routes.

File: app/controllers/home_controller.rb (hunk 5 above)
Comment (sudeeptarlekar) [reply app/controllers/home_controller.rb]: IMO, there will be two admins, `super-admin` and `organization-admin`, not sure though, hence I did not worked on any role related actions here.

File: config/routes.rb (hunk 6 above)
Comment (sudeeptarlekar) [reply config/routes.rb]: @rkr090 created issue for same https://github.com/bigbinary/acehelp/issues/82

**Comment by sudeeptarlekar** [comment]
@rkr090 _a Resolved PR comments. Please review.

**Pull request merged**